networking_cisco/plugins/cisco/device_manager/plugging_drivers/aci_vlan_trunking_driver.py

```diff
@@ -123,38 +122,34 @@ def get_vrf_context(self):
         else:
             return None
 
-    @property
-    def get_vrf_details(self):
-        if self.apic_driver:
-            return self._get_vrf_details
-        else:
-            return None
-
     def _get_vrf_context_gbp(self, context, router_id, port_db):
         l2p = self.apic_driver._network_id_to_l2p(
             context, port_db['network_id'])
+        # TODO(tbachman): should we just use the l3_policy_id
+        # from the l2p?
         l3p = self.apic_driver.gbp_plugin.get_l3_policy(
             context, l2p['l3_policy_id'])
-        return {'vrf_id': l3p['id']}
-
-    def _get_vrf_context_neutron(self, context, router_id, port_db):
-        return {'router_id': router_id}
-
-    def _get_vrf_details_gbp(self, context, **kwargs):
+        kwargs = {'vrf_id': l3p['id']}
         details = self.apic_driver.get_vrf_details(context, **kwargs)
         # The L3 out VLAN allocation uses UUIDs instead of names
+        details['vrf_id'] = details['l3_policy_id']
         details['vrf_name'] = details['l3_policy_id']
         # get rid of VRF tenant -- not needed  for GBP
         if details.get('vrf_tenant'):
             details['vrf_tenant'] = None
         return details
 
-    def _get_vrf_details_neutron(self, context, **kwargs):
-        router = self.l3_plugin.get_router(context, kwargs['router_id'])
+    def _get_vrf_context_neutron(self, context, router_id, port_db):
+        router = self.l3_plugin.get_router(context, router_id)
         vrf_info = self.apic_driver.get_router_vrf_and_tenant(router)
-        vrf_info['vrf_name'] = vrf_info['aci_name']
-        vrf_info['vrf_tenant'] = vrf_info['aci_tenant']
-        return vrf_info
+        details = {}
+        if self.apic_driver.per_tenant_context:
```

Comment:
The above produce names, and the VRF names have limited length. I'm using UUIDs here (except for per_tenant_context = False) in order to better handle uniqueness with limited characters.